src/framework/extensions/internal/scriptengine.cpp

```diff
@@ -275,6 +277,11 @@ RetVal<QJSValue> ScriptEngine::evaluateContent(const QByteArray& fileContent, co
     return rv;
 }
 
+const modularity::ContextPtr& ScriptEngine::iocContext() const
```

Comment:
You often pass the ptr by const reference, but I seem to recall I heard once that using (const) references for small things like (smart) pointers might make things slower. Is there any truth in that? And is there a specific reason you choose to use const reference here, or does it not really matter?

Reply:
There are no particular technical reasons (although the smart has some data that is copied)
The main reason I do this is to semantically indicate that this context is only read and not modified. 
Although technically, it is not protected from modification.

So, I would keep these const references to semantically indicate that it is read-only, and perhaps later actually protect it from modification.